src/main/java/Application/OpenCVServer/OpenCVServerOperator.java

```diff
@@ -0,0 +1,164 @@
+package Application.OpenCVServer;
+
+import java.io.*;
+import java.util.Arrays;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Vector;
+import java.net.*;
+
+import javax.imageio.ImageIO;
+import java.awt.image.BufferedImage;
+
+import Application.Utilities.DetectMarker;
+import Application.Utilities.HeartBeatRunner;
+import Application.Utilities.ImageProcessor;
+import Application.Utilities.OpenCVUtil;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import org.opencv.core.Mat;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+public class OpenCVServerOperator {
+    private ServerSocket serverSocket;
+    private Socket clientSocket;
+    private DataInputStream in;
+    private PrintWriter out;
+    private DetectMarker detector;
+    private int port;
+
+    private String fileName;
+
+    private BufferedImage currentImage;
+
+    public ServerRunner serverRunner;
+    public HeartBeatRunner beatRunner;
+
+    private static OpenCVServerOperator instance = null;
+
+    public OpenCVServerOperator() {
+        fileName= "read.png";
+    }
+
+    public void startConnection() throws IOException {
+        serverSocket = new ServerSocket(port);
+        clientSocket = serverSocket.accept();
+        out = new PrintWriter(clientSocket.getOutputStream(), true);
+        in = new DataInputStream(clientSocket.getInputStream());
+    }
```

Comment:
Maybe separate this behavior? Because this is also called in a loop, and in this loop:
- You want `.accept()` to be called every time because this would block until a connection is made
- You **DONT** want to create the server socket every time in the same port (just create it once)